src/contracts/deployment/BridgedTokenDeployer.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import "../vendored/mixins/BridgedTokensRegistry.sol";
+import "../CowProtocolVirtualToken.sol";
+
+/// @dev Contract contains the logic for deploying CowProtocolVirtualToken on gnosis chain
+/// This contract is needed, as the bridged CowToken address needs to be read from
+/// bridge contract at deployment time. After the deployment is done,
+/// the contract is no longer relevant
+/// Only some claim types are available on gnosis chain, which means that some parameters
+/// will not be used in the deployment.
+/// @author CoW Protocol Developers
+contract BridgedTokenDeployer {
+    /// @dev The token address of the CowToken on the Ethereum chain
+    address public immutable foreignToken;
+    /// @dev Multi Token Mediator from the ombi-bridge. It is used to get the
+    /// the address of the bridged CowToken
+    address public immutable multiTokenMediator;
+    /// @dev The merkle root of the Merkle tree from all claims for the gnosis chain
+    bytes32 public immutable merkleRoot;
```

Comment:
After this parameter, they all are the same as in the virtual token. Maybe we can add a newline and an extra comment that says that they are the same as in the virtual token.